### PR TITLE
Task 04 Directives and pipes

### PR DESCRIPTION
**Pull request by ShevtsovAnton**

### CODE REVIEW

File: src/app/features/courses/components/courses-item/courses-item.component.scss [hunk 1]
```diff
@@ -31,8 +32,16 @@
         box-sizing: border-box;
     }
 
+    &__body.top-rated{
+        background-color: red;
```
Comment (vnovikov-clgx) [inline src/app/features/courses/components/courses-item/courses-item.component.scss]: minor: If you use variables for colors, use in all app them

File: src/app/features/courses/components/search/search.component.ts [hunk 2]
```diff
@@ -7,9 +7,13 @@ import { Component } from '@angular/core';
 })
 export class SearchComponent {
 
+  @Output() makeSearchQuery: EventEmitter<string> = new EventEmitter();
   searchQuery = '';
 
-  search() {
-    console.log(this.searchQuery);
+  search(event) {
```
Comment (vnovikov-clgx) [inline src/app/features/courses/components/search/search.component.ts]: Pls add type here, and also i not sure, but `keypress` haven't type `click`. Pls check it

File: src/app/features/courses/containers/courses-page/courses-page.component.ts [hunk 3]
```diff
@@ -1,89 +1,128 @@
...
         description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
-        imagePath: '../../../../assets/node_new.jpeg'
+        imagePath: '../../../../assets/node_new.jpeg',
+        topRated: false
       }
     ];
+    this.filteredCoursesList = [...this.coursesList];
+  }
+
+  private toLowerCase(item: string) {
```
Comment (vnovikov-clgx) [inline src/app/features/courses/containers/courses-page/courses-page.component.ts]: minor: add return type

File: src/app/features/courses/pipes/order-by.pipe.mock.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+@Pipe({
+  name: 'orderBy'
+})
+
+export class OrderByMockPipe implements PipeTransform {
+    transform(value: CoursesItemModel[]): CoursesItemModel[] {
+        return [];
+    }
+}
```
Comment (vnovikov-clgx) [inline src/app/features/courses/pipes/order-by.pipe.mock.ts]: Where are you use it?

File: src/app/features/courses/pipes/minutes-to-hours.pipe.mock.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({name: 'minutesToHours'})
+export class MinutesToHoursMockPipe implements PipeTransform {
+    transform(value: number): string {
+        return '';
+    }
+}
```
Comment (vnovikov-clgx) [inline src/app/features/courses/pipes/minutes-to-hours.pipe.mock.ts]: Where are you use it?

File: src/app/features/courses/components/courses-item/courses-item.component.scss (hunk 1 above)
Comment (ShevtsovAnton) [reply src/app/features/courses/components/courses-item/courses-item.component.scss]: fixed: moved colors to variables.

File: src/app/features/courses/components/search/search.component.ts (hunk 2 above)
Comment (ShevtsovAnton) [reply src/app/features/courses/components/search/search.component.ts]: The idea was to make search possible when clicking on search icon or when pressing 'Enter' button,  for better user experience. 
The search method was called when  search icon was clicked and when 'enter' key was pressed in the input field.
Made the following changes: 
- click on search icon calls search method;
- 'enter' key press calls handle keypress method, which checks, if 'Enter' was pressed.  
Event types added, where required.

File: src/app/features/courses/containers/courses-page/courses-page.component.ts (hunk 3 above)
Comment (ShevtsovAnton) [reply src/app/features/courses/containers/courses-page/courses-page.component.ts]: added return type.

File: src/app/features/courses/pipes/order-by.pipe.mock.ts (hunk 4 above)
Comment (ShevtsovAnton) [reply src/app/features/courses/pipes/order-by.pipe.mock.ts]: The pipe is used in the courses page component template.
the mock is used in the following spec: 
src\app\features\courses\containers\courses-page\courses-page.component.spec.ts

File: src/app/features/courses/pipes/minutes-to-hours.pipe.mock.ts (hunk 5 above)
Comment (ShevtsovAnton) [reply src/app/features/courses/pipes/minutes-to-hours.pipe.mock.ts]: The pipe is used in the courses item component template.
the mock is used in the following spec: 
src\app\features\courses\components\courses-item\courses-item.component.spec.ts

File: src/app/features/courses/components/search/search.component.ts (hunk 2 above)
Comment (vnovikov-clgx) [reply src/app/features/courses/components/search/search.component.ts]: you can add few types here

**Pull request merged**